### PR TITLE
APF-1594-Updated spring boot to 2.0.5 and other dependencies version

### PR DESCRIPTION
**Pull request by SudharsanThumattiSathyamoorthy**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -151,19 +151,19 @@
...
                 <artifactId>mockito-core</artifactId>
-                <version>2.18.3</version>
+                <version>2.22.0</version>
                 <scope>test</scope>
             </dependency>
             <dependency>
                 <groupId>org.mockito</groupId>
                 <artifactId>mockito-junit-jupiter</artifactId>
-                <version>2.18.3</version>
+                <version>2.20.1</version>
```
Comment (SudharsanThumattiSathyamoorthy) [inline pom.xml]: The latest version for ```mockito-junit-jupiter``` is ```2.22.0```. But the latest version seems to have bug. When I updated the version  to ```2.22.0```, I got the below exception. They have raised the issue in junit forum.

```
https://github.com/junit-team/junit5/issues/833
```

```
Tests run: 6, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.318 sec <<< FAILURE! - in com.vmware.connectors.aws.cert.AwsCertControllerMockedTest
testApprove{String, int}[1]  Time elapsed: 0.022 sec  <<< ERROR!
org.junit.jupiter.api.extension.ParameterResolutionException: Failed to resolve parameter [java.lang.String approvalUrl] in executable [void com.vmware.connectors.aws.cert.AwsCertControllerMockedTest.testApprove(java.lang.String,int)]
Caused by: java.lang.AbstractMethodError

```

File: pom.xml [hunk 2]
```diff
@@ -47,12 +47,12 @@
             <dependency>
                 <groupId>javax.xml.bind</groupId>
                 <artifactId>jaxb-api</artifactId>
-                <version>2.3.0</version>
+                <version>2.4.0-b180830.0359</version>
```
Comment (rworsnop) [inline pom.xml]: They have a weird release system. Even though it's in maven central, this appears to be a pre-release candidate. I did a bit of detective work [here](https://github.com/javaee/jaxb-v2/releases).

File: pom.xml (hunk 1 above)
Comment (rworsnop) [reply pom.xml]: Please try updating the maven-surefire-plugin to 2.22.0, removing the dependencies from it, then adding junit-jupiter-engine under dependencyManagement. See https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html

File: pom.xml (hunk 2 above)
Comment (SudharsanThumattiSathyamoorthy) [reply pom.xml]: The latest stable version is ```2.3.1```, but in the maven repo the latest stable available is ```2.3.0```. I have updated the same version.

File: pom.xml [hunk 3]
```diff
@@ -151,21 +151,26 @@
...
                 <artifactId>mockito-junit-jupiter</artifactId>
-                <version>2.18.3</version>
+                <version>2.22.0</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-engine</artifactId>
+                <version>5.3.0</version>
+            </dependency>
```
Comment (rworsnop) [inline pom.xml]: Test scope.

File: pom.xml [hunk 4]
```diff
@@ -189,17 +194,12 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>2.19.1</version>
+                    <version>2.22.0</version>
                     <dependencies>
                         <dependency>
                             <groupId>org.junit.platform</groupId>
```
Comment (rworsnop) [inline pom.xml]: You should be able to get rid of junit-platform-surefire-provider. https://github.com/junit-team/junit5/issues/1536

Comment (SudharsanThumattiSathyamoorthy) [reply pom.xml]: Removed ```junit-platform-surefire-provider``` dependency from ```maven-surefire-plugin```.

File: pom.xml (hunk 3 above)
Comment (rworsnop) [reply pom.xml]: This should have test scope.

**Pull request merged**